cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/MultiOpsPostgresSinkConsumerIT.java

```diff
@@ -0,0 +1,230 @@
+        TestHelper.deleteTopicInKafka(kafkaContainerIp, kafkaContainer.KAFKA_PORT,
+                Arrays.asList("dbserver1.public.test_table"));
+
+        // Drop the table in YugabyteDB as well as the sink table in Postgres
+        TestHelper.execute(DROP_TABLE_STMT);
+
+        // Using the same drop table statement since the table name is the same for both YugabyteDB
+        // and Postgres
+        TestHelper.executeInPostgres(pgContainerIp, DROP_TABLE_STMT);
+    }
+
+    @AfterAll
+    public static void afterAll() throws Exception {
+        // Stop the running containers
+        pgContainer.stop();
+        kafkaConnectContainer.stop();
+        kafkaContainer.stop();
+    }
+
+    @Test
+    public void testInsertUpdateDelete() throws Exception {
+        // At this point in time, all the containers are up and running properly
+        TestHelper.execute(String.format(INSERT_FORMAT_STRING, 1, "Vaibhav", "Kushwaha", 23.456));
+        TestHelper.execute("UPDATE test_table SET last_name='Kush' WHERE id = 1;");
+        TestHelper.execute("DELETE FROM test_table WHERE id = 1;");
+
+        // Wait some time for the table to get created in postgres and for replication to complete
+        Thread.sleep(5000);
+
+        TestHelper.assertRecordCountInPostgres(0, pgContainerIp);
+    }
+
+    @Test
+    public void testUpdateAfterInsert() throws Exception {
+        TestHelper.execute(String.format(INSERT_FORMAT_STRING, 1, "Vaibhav", "Kushwaha", 23.456));
+        TestHelper.execute("UPDATE test_table SET last_name='Kush' WHERE id = 1;");
+
+        // Wait some time for the table to get created in postgres and for replication to complete
+        Thread.sleep(5000);
+
+        ResultSet rs = TestHelper.executeAndGetResultSetPostgres(pgContainerIp,
+                "SELECT * FROM test_table;");
+        if (rs.next()) {
+            // Expect one row with the updated value of last_name
+            assertEquals("Kush", rs.getString("last_name"));
+        }
+        else {
+            // Fail in case no ResultSet is retrieved from the query
+            fail();
+        }
+    }
+
+    @Disabled
+    @Test
+    public void testBatchInserts() throws Exception {
+        int totalRowsToBeInserted = 50;
+        try (Connection conn = TestHelper.getConnectionOnYugabyteDB()) {
+            Statement st = conn.createStatement();
+            int ind = 0;
+            while (ind < totalRowsToBeInserted) {
+                for (int i = ind; i < ind + 5; ++i) {
+                    st.addBatch(String.format(INSERT_FORMAT_STRING, i, "first_" + i, "last_" + i, 23.456));
+                }
+                st.executeBatch();
+                ind += 5;
+
+                // Clear batch for next iteration
+                st.clearBatch();
+            }
+        }
+        catch (SQLException e) {
+            throw e;
+        }
+
+        // Wait for records to be replicated across Postgres
+        Thread.sleep(5000);
```

Comment:
Another simple option is to use `TestHelper.assertRecordCountInPostgres(totalRowsToBeInserted, pgContainerIp);` within Awaitility fn. here.

Reply:
I am doing that as a part of my next refactoring PR.